mainnet-contracts/src/PufferVaultV3.sol

```diff
@@ -0,0 +1,233 @@
+
+        if (params.rewardsAmount > $.allowedRewardMintAmount) {
+            revert InvalidMintAmount();
+        }
+
+        if (($.lastRewardMintTimestamp + $.allowedRewardMintFrequency) > block.timestamp) {
+            revert NotAllowedMintFrequency();
+        }
+
+        BridgeData memory bridgeData = $.bridges[params.bridge];
+
+        if (bridgeData.destinationDomainId == 0) {
+            revert BridgeNotAllowlisted();
+        }
+
+        uint256 ethToPufETHRate = convertToShares(1 ether);
+        uint256 shares = ethToPufETHRate.mulDiv(params.rewardsAmount, 1 ether, Math.Rounding.Floor);
+
+        $.lastRewardMintTimestamp = uint40(block.timestamp);
+        $.totalRewardMintAmount += uint104(params.rewardsAmount);
+
+        _mint(address(this), shares);
+        _approve(address(this), address(LOCKBOX), shares);
+        LOCKBOX.deposit(shares);
+
+        // This contract approves transfer to the bridge
+        XPUFETH.approve(address(params.bridge), shares);
+
+        MintAndBridgeData memory bridgingCalldata = MintAndBridgeData({
+            rewardsAmount: uint128(params.rewardsAmount),
+            ethToPufETHRate: uint128(ethToPufETHRate),
+            startEpoch: params.startEpoch,
+            endEpoch: params.endEpoch,
+            rewardsRoot: params.rewardsRoot,
+            rewardsURI: params.rewardsURI
+        });
+
+        BridgingParams memory bridgingParams =
+            BridgingParams({ bridgingType: BridgingType.MintAndBridge, data: abi.encode(bridgingCalldata) });
+
+        // Encode data for the target contract call
+        bytes memory encodedData = abi.encode(bridgingParams);
+
+        IBridgeInterface(params.bridge).xcall{ value: msg.value }({
+            destination: bridgeData.destinationDomainId, // Domain ID of the destination chain
+            to: L2_REWARD_MANAGER, // Address of the target contract
+            asset: address(XPUFETH), // Address of the token contract
+            delegate: msg.sender, // Address that can revert or forceLocal on destination
+            amount: shares, // Amount of tokens to transfer
+            slippage: 0, // Max slippage the user will accept in BPS (e.g. 300 = 3%)
+            callData: encodedData // Encoded data to send
+         });
+
+        emit MintedAndBridgedRewards({
+            rewardsAmount: params.rewardsAmount,
+            startEpoch: params.startEpoch,
+            endEpoch: params.endEpoch,
+            rewardsRoot: params.rewardsRoot,
+            ethToPufETHRate: ethToPufETHRate,
+            rewardsURI: params.rewardsURI
+        });
+    }
+
+    /**
+     * @notice Sets the L2 reward claimer.
+     * @param bridge The address of the bridge.
+     * @param claimer The address of the new claimer.
+     * @dev Restricted in this context is like the `whenNotPaused` modifier from Pausable.sol
+     */
+    function setL2RewardClaimer(address bridge, address claimer) external payable restricted {
```

Comment:
What is the benefit of having a bridging tx that sets the claimer on L2? Why not a tx on L2 to do the same?